eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/DependencyComparator.java

```diff
@@ -0,0 +1,55 @@
+package com.azure.tools.bomgenerator;
+
+import org.apache.maven.model.Dependency;
+
+import java.util.Comparator;
+
+public class DependencyComparator implements Comparator {
```

Comment:
Why isn't the `Comparator` typed with the generic it is using? Also, given how simple this comparator is, does this class need to exist?

Reply:
So, the issue I hit into was that by default java compares references while finding an element in the set.

So, say I have two instaces of dependencies -> dep1, dep2 they have identical metadata but point to different references at runtime - set can only find dep2 in the set if dep2 was added. Comparator helped me do a different match than the actual reference.

I could use a lambda in the treeset creation but I will have to do it at 3 places - so thought creating a new type might be better here.

What other way would you suggest I use? Thanks!

Reply:
Given this is implementation, I think we could turn the Comparators into static lambdas for the time being to simplify the package structure.

Another option would have the classes implement `equals` or `hashCode` which would help perform the same functionality as the default comparator. Or having the types implement Comparable.

Reply:
Removed the type.